File: benchmarks/pontos.c
#include <stdio.h>

typedef struct { int x; int y; } Ponto;

int distanciaManhattan(Ponto* p) {
    int dx = p->x < 0 ? -p->x : p->x;
    int dy = p->y < 0 ? -p->y : p->y;
    return dx + dy;
}

int main() {
    int total = 0;
    for (int i = 0; i < 1000000; i++) {
        Ponto p = { .x = (i * 7 + 3) % 1000 - 500, .y = (i * 13 + 7) % 1000 - 500 };
        total += distanciaManhattan(&p);
    }
    printf("%d\n", total);
    return 0;
}

File: benchmarks/pontos.rs
struct Ponto { x: i32, y: i32 }

impl Ponto {
    fn distancia_manhattan(&self) -> i32 {
        self.x.abs() + self.y.abs()
    }
}

fn main() {
    let mut total: i32 = 0;
    for i in 0..1_000_000i32 {
        let p = Ponto { x: (i * 7 + 3) % 1000 - 500, y: (i * 13 + 7) % 1000 - 500 };
        total += p.distancia_manhattan();
    }
    println!("{}", total);
}
